pytential/symbolic/compiler.py

```diff
@@ -585,20 +585,19 @@ def map_int_g(self, expr, name_hint=None):
         try:
             return self.expr_to_var[expr]
         except KeyError:
+            from pytential.utils import sort_arrays_together
+            densities, source_kernels = \
+                    *sort_arrays_together(expr.densities, expr.source_kernels)
```

Comment:
I'm honestly surprised this succeeds--what is it sorting by? [For the target derivatives](https://github.com/inducer/pytential/blob/b6e627ef815c740324e553615b2f4e21292b39ae/pytential/symbolic/compiler.py#L577), I ended up using `repr` as the sort key.